libsolidity/codegen/CompilerContext.cpp

```diff
@@ -57,6 +60,51 @@ void CompilerContext::startFunction(Declaration const& _function)
 	*this << functionEntryLabel(_function);
 }
 
+void CompilerContext::callLowLevelFunction(
+	string const& _name,
+	unsigned _inArgs,
+	unsigned _outArgs,
+	function<void(CompilerContext&)> const& _generator
+)
+{
+	eth::AssemblyItem retTag = pushNewTag();
+	CompilerUtils(*this).moveIntoStack(_inArgs);
+
+	auto it = m_lowLevelFunctions.find(_name);
+	if (it == m_lowLevelFunctions.end())
+	{
+		eth::AssemblyItem tag = newTag().pushTag();
+		m_lowLevelFunctions.insert(make_pair(_name, tag));
+		m_lowLevelFunctionGenerationQueue.push(make_tuple(_name, _inArgs, _outArgs, _generator));
+		*this << tag;
+	}
+	else
+		*this << it->second;
+	appendJump(eth::AssemblyItem::JumpType::IntoFunction);
+	adjustStackOffset(_outArgs - 1 - _inArgs);
```

Comment:
The subtraction results in `unsigned` (because `unsigned` - `int` is an `unsigned`) but `adjustStackOffset` wants an `int`.  The conversion from `unsigned` to `int` is undefined on numbers larger than `INT_MAX`.